### PR TITLE
updates for -c args.

### PR DESCRIPTION
**Pull request by gknisely**
When a user specifies the columns with the -c arg, should use those columns for the inserts.

For example, I have a table that I created before running termsql.  

DROP TABLE IF EXISTS "shapes_tmp";
CREATE TABLE "shapes_tmp"
(
  shape_key integer primary key autoincrement,
  shape_id text,
  shape_pt_lat double precision,
  shape_pt_lon double precision,
  shape_pt_sequence integer,
  shape_dist_traveled double precision
);

And if I run termsql with the -a and -c option, it will not work as sqlite3 sees shape_key as an extra column.  (notice no shape_key column in the cmd) 

termsql -a -i ./shapes.txt -c 'shape_id,shape_pt_lat,shape_pt_lon,shape_pt_sequence,shape_dist_traveled' -1 -d '|' -t shapes_tmp -o transit.sqlite

Traceback (most recent call last):
  File "./termsql.old/termsql", line 369, in <module>
    c.executemany('insert into '+table_name+' values ('+placeholder_str+')', inserts)
sqlite3.OperationalError: table shapes_tmp has 6 columns but 5 values were supplied

insert into '+table_name+'('+cols+') values ('+placeholder_str+')' will fix this issue.  

Basically performing:  
    INSERT INTO table_name (column1, column2, column3,...)
    VALUES (value1, value2, value3,...)

### CODE REVIEW

**Comment by tobimensch** [comment]
Hi gknisely,

thanks for reporting this and for supplying a fix. Will look into this today.

Your fix looks correct at first glance, but I'll have to look if there are any side issues,
regressions etc.

**Comment by tobimensch** [comment]
btw. seeing that you have prefixed all column names with shape_ makes me wonder if introducing a feature for defining a prefix makes sense. I mean it would save you from writing shape_ over and over, but that might be feature bloat.

**Comment by tobimensch** [comment]
Ok, I looked into your fix some more. In its current form it's probably breaking a few things. The +'('+cols+')  part has to have at least as many fields as the ('+placeholder_str+')' part has placeholders for fields (which in your example is true but only because you specify the right amount of custom column names), which is equal to the amount of total fields of the input data. This also means we can eradicate your special case and make it more generalized, meaning we don't need the "if" condition that you introduced.

**Comment by tobimensch** [comment]
I think I have solved this with the latest commit. I can close this request now. Please try it.

https://github.com/tobimensch/termsql/commit/2c675d2db4569363b96881e68865028d97ee5062
